src/engines/sharp.js

```diff
@@ -64,17 +70,35 @@ module.exports = {
         if (options.sequentialRead) {
             sharpInstance = sharpInstance.sequentialRead();
         }
+
+        var operationsForExecution = [];
         operations.forEach(function (operation) {
+            var name = operation.name;
             var args = operation.args;
             if (operation.name === 'resize' && typeof pipeline.options.maxOutputPixels === 'number' && args[0] * args[1] > pipeline.options.maxOutputPixels) {
                 throw new errors.OutputDimensionsExceeded('resize: Target dimensions of ' + args[0] + 'x' + args[1] + ' exceed maxOutputPixels (' + pipeline.options.maxOutputPixels + ')');
             }
+            // in sharp crop is implemented as options to resize
+            if (operation.name === 'crop') {
+                name = 'resize';
+                args = [ null, null, { fit: 'cover', position: args[0] } ];
+
+                const locatedOperation = locatePreviousCommand(operationsForExecution, name);
```

Comment:
Since crop and resize are now one and the same, this `locatePreviousCommand` stuff tries to compact the the commands back together - this can probably be removed for the purposes of removing any chance at functional change.

Reply:
Unless it's important for you to maintain backwards compatibility with express-processimage I think we can also just align with the new sharp api? At least if we find it more elegant :)

Reply:
Ahhh right yeah - need to remember that this isn’t necessarily meant to abstract the specific engine away but rather select the appropriate engine..

I’ve noticed this elsewhere but there’s obviously a tension between generic commands meaning the right engine is picked for the job versus specific commands that might expose more but tie the choices. My feeling is a handful of generic commands might be useful - particularly setFormat which became a big win in express-processimage but I’ve not thought it through fully.